src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -492,6 +492,26 @@ parseNumberBinder = NumberBinder <$> (sign <*> number)
          <|> (symbol' "+" >> return id)
          <|> return id
 
+parseTypedBinder :: TokenParser Binder
+parseTypedBinder = do
+  b <- typeableBinder
+  indented *> doubleColon
+  t <- parseType
+  return $ TypedBinder t b
+    where typeableBinder = P.choice (map P.try
```

Comment:
Can you give an example where this pattern is followed somewhere else in the codebase (I'm not sure where to start restructuring this, but I want to do it the right way)